### PR TITLE
Fix YAML/Cython tests

### PR DESCRIPTION
**Pull request by bryanwweber**
These changes fix a few things that broke when testing the installed Cython interface (via `python -m unittest -v cantera.test`)

Changes proposed in this pull request:
- Install the `test/data/test_subdir` in the Cython interface
- Avoid using paths in the test files that implicitly relied on being in the source tree
- Write test conversions into the `test_work_dir` instead of the cwd

### CODE REVIEW

File: interfaces/cython/cantera/test/test_convert.py
Comment (speth) [inline interfaces/cython/cantera/test/test_convert.py:570]: If we're going all in on the use of `pathlib`, why don't we just go ahead and convert the `cantera_data`, `test_work_dir`, and `test_data_dir` variables to instances of `pathlib.Path`?

**Comment by speth** [comment]
Don't forget to `rebase -i` so that your fixup commit gets squashed correctly (I'm assuming GitHub doesn't do `autosquash`).

File: interfaces/cython/cantera/test/test_convert.py
Comment (bryanwweber) [reply interfaces/cython/cantera/test/test_convert.py:570]: Sure. The test failures are because cti2yaml expects a string as the filename, so I'm working on making that pathlib compatible.

Comment (bryanwweber) [reply interfaces/cython/cantera/test/test_convert.py:570]: So at the moment, most of the other test cases assume that `cantera_data` et al. are strings, and pass them to various `os.path` functions that only work with strings in Python 3.5. 3.6 upgrades all the `os.path` functions to also work with pathlib-like objects, but since we're still supporting 3.5 and I don't think it makes sense to update all of the ck2cti tests that will be going away, I'm not going to change this for now.

**Comment by bryanwweber** [comment]
> Don't forget to `rebase -i` so that your fixup commit gets squashed correctly (I'm assuming GitHub doesn't do `autosquash`).

Once the changes are approved, I'll autosquash and then it can be merged.

**Comment by bryanwweber** [comment]
This is rebased and squashed now. :+1:?

**Comment by bryanwweber** [comment]
I'll merge this tomorrow if no further comments. This is blocking for Cantera/conda-recipes#19

**Review by speth — APPROVED** [review]

**Pull request merged**
